File: C++/Problems/BASICS/09_Square_Pattern.cpp
/*
Q1.Example:
    for n = 4
    1234
    1234
    1234
    1234
*/
// #include <iostream>
// using namespace std;
// int main() {
    
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;
    
//     for (int i=1; i<=n; i++){
//         for (int j=1; j<=n; j++){
//             cout<<j;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

/*
Q2. Example:
    ABCD
    ABCD
    ABCD
    ABCD
*/

// #include <iostream>
// using namespace std;
// int main() {
    
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;
    
//     for (int i=1; i<=n; i++){
//         char ch='A';
//         for (int j=1; j<=n; j++){
//             cout<<ch;
//             ch = ch+1;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

/*
Q3. for n = 3 Example:
    123
    456
    789
*/

// #include <iostream>
// using namespace std;
// int main() { 
//     int n,num;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     num=1;
//     for (int i=1; i<=n; i++){
//         for (int j=1 ; j<=n; j++){
//             cout<<num;
//             num++;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }


/*
Q4. for n = 3 Example:
    ABC
    DEF
    GHI
*/

// #include <iostream>
// using namespace std;
// int main() { 
//     int n;
//     char ch;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     ch ='A';
//     for (int i=1; i<=n; i++){
//         for (int j=1 ; j<=n; j++){
//             cout<<ch;
//             ch=ch+1;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

'''
Q5. For n = 4 Example : 
    *
    **
    ***
    ****
'''

// #include<iostream>
// using namespace std;
// int main(){
//     int n;
//     cout<< "Enter a Number : ";
//     cin>>n;

//     for (int i=1; i<=n; i++){
//         cout<<"*"*i<<"\n";
//     }
//     return 0;
// }

'''
Q6. for n = 4 Example:
    1
    22
    333
    4444
'''

// #include <iostream>
// using namespace std;
// int main() {
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     for(int i=1; i<=n; i++){
//         for(int j = 1; j<=i; j++){
//             cout<<i;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

'''
Q7. for n = 4 Example:
    A
    BB
    CCC
    DDDD
'''

// #include<iostream>
// using namespace std;
// int main(){
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     char ch = 'A';
//     for(int i=1; i<=n; i++){
//         for(int j=1;  j<=i; j++){
//             cout<<ch;
//         }
//         ch = ch +1;
//     }
//     return 0;
// }


'''
Q8. for n = 4 Example : 
    1
    21
    321
    4321
'''

// #include<iostream>
// using namespace std;
// int main(){
//     int n=4;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     for(int i=1; i<=n; i++){
//         for(int j=i; j>0; j--){
//             cout<<j;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

'''
Q9. for  n=3 Example :
    1
    23
    456
'''

// #include<iostream>
// using namespace std;
// int main(){
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     int count  = 1;
//     for(int i=1; i<=n; i++){
//         for(int j=1; j<=i; j++){
//             cout<<count;
//             count = count + 1;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

'''
Q10. for  n=4 Example :
    A
    BC
    DEF
    GHIJ
'''

// #include<iostream>
// using namespace std;
// int main(){
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;

//     char ch = 'A';

//     for (int i=1; i<=n; i++){
//         for(int j=1; j<=i; j++){
//             cout<<ch;
//             ch = ch + 1;
//         }
//         cout<<"\n";
//     }
//     return 0;
// }

'''
Q11 for n=4 Example :
    A
    BA
    CBA
    DCBA
'''

// #include <iostream>
// using namespace std;
// int main() {
//     int n;
//     cout<<"Enter a Number : ";
//     cin>>n;
//     char startChar = 'A';
//     for (int i = 1; i <= n; i++) {
//         for (int j = startChar; j >= 'A'; j--) {
//             cout << (char)j;
//         }
//         cout << "\n";
//         startChar++;
//     }
//     return 0;
// }

File: Python/Problems/BASICS/09_Square_Pattern.py
'''
Q1.Example:
    for n = 4
    1234
    1234
    1234
    1234
'''

# n =int(input("Enter a Number : "))
# for i in range(1,n+1):
#     for j in range (1,n+1):
#         print(j, end="")
#     print()

'''
Q2. for n = 4 Example:
    ABCD
    ABCD
    ABCD
    ABCD
'''

# n=int(input("Enter a Number : "))
# for i in range(1,n+1):
#     char ch = 'A'
#     for j in range(1,n+1):
#         print(ch)
#         ch = chr(ord(ch)+1) # here ord() converts character into there ASCII code while chr converts it into Character again and +1 is used to increment the character when it is converted into ASCII code 
#     print() # this is used to print the next line

'''
Q3. for n =3 Example: 
    123
    456
    789
'''

# n=int(input("Enter a Number : "))
# num = 1

# for i in range(1,n+1):
#     for j in range(1,n+1):
#         print(num,end="")
#         num = num +1
#     print()

'''
Q4. for n = 4 Example:
    ABC
    DEF
    GHI
'''

# n=int(input("Enter a Number : "))
# ch = 'A'

# for i in range(1,n+1):
#     for j in range(1,n+1):
#         print(ch,end="")
#         ch = chr(ord(ch)+1) 
#     print()

'''
Q5. For n = 4 Example : 
    *
    **
    ***
    ****
'''

# a = int(input("Enter a Number : "))

# for i in range(1,a+1):
#     print("*"*i)

'''
Q6. for n = 4 Example:
    1
    22
    333
    4444
'''

# n = int(input("Enter a Number : "))

# for i in range(1,n+1):
#     for j in range(1,i+1):
#         print(i,end="")
#     print()

'''
Q7. for n = 4 Example:
    A
    BB
    CCC
    DDDD
'''
# n = int(input("Enter a Number : "))
# ch = "A"
# for i in range(1,n+1):
#     for j in range(1,i+1):
#         print(ch,end="")
#     ch = chr(ord(ch)+1) # we kept this outside the inner loop because we want to print the same character in the same row
#     print()

'''
Q8. for n = 4 Example : 
    1
    21
    321
    4321
'''

# n = int(input("Enter a Number : "))

# for i in range(1,n+1):
#     for j in range(i, 0, -1):
#         print(j,end="")
#     print()

'''
Q9. for  n=4 Example :
    1
    23
    456
    78910
'''

# n = int(input("Enter a Number : "))
# count = 1
# for i in range(1,n+1):
#     for j in range(1,i+1):
#         print(count, end="")
#         count = count + 1
#     print()

'''
Q10. for  n=4 Example :
    A
    BC
    DEF
    GHIJ
'''

# n = int(input("Enter a Number : "))
# ch = 'A'
# for i in range(1,n+1):    
#     for j in range(1,i+1):
#         print(ch,end="")
#         ch= chr(ord(ch)+1)
#     print()


'''
Q11 for n=4 Example :
    A
    BA
    CBA
    DCBA
'''

# n = int(input("Enter a Number : "))
# ch = ord('A')

# for i in range(n):
#     for j in range(ch, ord('A') - 1, -1):
#         print(chr(j), end="")
#     print()
#     ch += 1


